CMakeLists.txt

```diff
@@ -16,7 +16,7 @@
 
 cmake_minimum_required(VERSION 3.0.2)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-comment")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-comment -mf16c")
```

Comment:
Is this flag related to CentOS?  It seems more related to CPU architecture.

And if we're actually trying to make the best use of the CPU, would it make sense to instead use `-march=native`?

Reply:
The issue shows up on CentOS, but, in general, it's for 3rdparty/mshadow definitions that we use.